GDAXSharp/WebSocket/WebSocket.cs

```diff
@@ -0,0 +1,135 @@
+
+        private CancellationToken CancellationToken { get; set; }
+        private ProductType[] ProductTypes { get; set; }
+        private WebSocket4Net.WebSocket WebSocketFeed { get; set; }
+
+        public WebSocket(IAuthenticator authenticator, IClock clock, CancellationToken cancellationToken, bool sandBox = false) : base(authenticator, clock, sandBox)
+        {
+            CancellationToken = cancellationToken;
+        }
+
+        public void GetTickerChannel(params ProductType[] productTypes)
+        {
+            ProductTypes = productTypes;
+            if (ProductTypes.Length == 0)
+            {
+                throw new ArgumentException("You must specify at least one product type");
+            }
+
+            var socketUrl = SandBox ? SandBoxApiUri : ApiUri;
+
+            WebSocketFeed = new WebSocket4Net.WebSocket(socketUrl);
+            WebSocketFeed.Closed += WebSocket_Closed;
+            WebSocketFeed.Error += WebSocket_Error;
+            WebSocketFeed.MessageReceived += WebSocket_MessageReceived;
+            WebSocketFeed.Opened += WebSocket_Opened;
+            WebSocketFeed.Open();
+        }
+
+        private void WebSocket_Opened(object sender, EventArgs e)
+        {
+            var timeStamp = Clock.GetTime().ToTimeStamp();
+            var json = JsonConvert.SerializeObject(new TickerChannel
+            {
+                Type = "subscribe",
+                ProductIds = ProductTypes,
+                Channels = new List<Channel>
+                {
+                    new Channel
+                    {
+                        Name = "ticker",
+                        ProductIds = ProductTypes
+                    },
+                    new Channel
+                    {
+                        Name = "level2",
+                        ProductIds = ProductTypes
+                    },
+                    new Channel
+                    {
+                        Name = "user",
+
+                        ProductIds = ProductTypes
+                    }
+                },
+                Timestamp = timeStamp.ToString("F0", CultureInfo.InvariantCulture),
+                Key = Authenticator.ApiKey,
+                Passphrase = Authenticator.Passphrase,
+                Signature = ComputeSignature(HttpMethod.Get, Authenticator.UnsignedSignature, timeStamp, "/users/self/verify", null)
+            });
+
+            WebSocketFeed.Send(json);
+        }
+
+        private void WebSocket_MessageReceived(object sender, MessageReceivedEventArgs e)
+        {
+            if (CancellationToken.IsCancellationRequested)
+            {
+                WebSocketFeed.Close();
+                return;
+            }
+
+            var json = e.Message;
+            var response = JsonConvert.DeserializeObject<BaseMessage>(json);
+
+            switch (response.Type)
+            {
+                case ResponseType.Subscriptions:
+                    // TODO: Implement this and more?
```

Comment:
what is this for? if you aren't going to add this here i would remove and add in a subsequent PR

Reply:
There are more response types, so I was thinking about adding different `EventHandler` for each `ResponseType`.

Reply:
awesome. just remove the TODO if you aren't going to implement it yet.